pkg/daemon/daemon.go

```diff
@@ -990,11 +994,6 @@ func (dn *Daemon) checkStateOnFirstRun() error {
 
 	// Bootstrapping state is when we have the node annotations file
 	if state.bootstrapping {
```

Comment:
i.e. this runs only at cluster installation

Reply:
Oh duh, yep.

We should be able to actually test this too by injecting a `MachineConfig` with the `reboot.service` in question, I was lazy the first time and didn't do that but it's probably worth doing now to avoid another potential QE cycle?

Or it should replicate by just - on a cluster with this pr, do `mv /etc/ignition-machine-config-encapsulated.{bak,json}` and kill the MCD pod.

Reply:
> We should be able to actually test this too by injecting a MachineConfig with the reboot.service in question

So the test would be:

- spin up a 4.4 cluster with a MachineConfig with restart.service
- upgrade to a 4.5/4.6 that does have this fix

right? I'm just worried about the fact the bug doesn't happen 100% of the times and we would waste everybody's cycles

Reply:
I mean, if the intention of 1822 was this fix but this is just fixing the actual call to the hack, I think we can merge this and let QE test again right?